### DISCUSSION TITLE
how to integrate a custom websocket server?

### DISCUSSION THREAD

**Discussion by pbreah** [wailsapp/wails#2066] [General]
I am aware that websockets are available in dev mode as that is how the cli and app connect. But there is nothing in the docs on how to do this for production apps. Is this possible?

**Comment by stffabi** [comment]
Unfortunately it's not possible to use websockets for communication between backend and frontend in production mode. At least not without spawning your own http server listening on localhost, which I wouldn't recommend for a production build.
There's updated docs coming with the next release, where we at least state it's not supported: https://wails.io/docs/next/reference/options#assetserver

If you are interested in some more technical details why it's not supported, there's a thread on the slack channel with some more information. Just search for websockets and you should be able to find it.

**Reply by gerbil** [reply]
> If you are interested in some more technical details why it's not supported, there's a thread on the slack channel with some more information. Just search for websockets and you should be able to find it.

Could you please paste it here?
